aspnetcore/mvc/controllers/testing/sample/TestingControllersSample/tests/TestingControllersSample.Tests/UnitTests/ApiIdeasControllerTests.cs

```diff
@@ -117,6 +127,130 @@ public async Task ForSession_ReturnsIdeasForSession()
             var idea = returnValue.FirstOrDefault();
             Assert.Equal("One", idea.Name);
         }
+        #endregion
+
+        #region snippet_ForSessionActionResult_ReturnsNotFoundObjectResultForNonexistentSession
+        [Fact]
+        public async Task ForSessionActionResult_ReturnsNotFoundObjectResultForNonexistentSession()
+        {
+            // Arrange
+            var mockRepo = new Mock<IBrainstormSessionRepository>();
+            var controller = new IdeasController(mockRepo.Object);
+            var nonExistentSessionId = 999;
+
+            // Act
+            var result = await controller.ForSessionActionResult(nonExistentSessionId);
+
+            // Assert
+            var actionResult = Assert.IsType<ActionResult<List<IdeaDTO>>>(result);
+            Assert.IsType<NotFoundObjectResult>(actionResult.Result);
+        }
+        #endregion
+
+        #region snippet_ForSessionActionResult_ReturnsIdeasForSession
+        [Fact]
+        public async Task ForSessionActionResult_ReturnsIdeasForSession()
+        {
+            // Arrange
+            int testSessionId = 123;
+            var mockRepo = new Mock<IBrainstormSessionRepository>();
+            mockRepo.Setup(repo => repo.GetByIdAsync(testSessionId))
+                .Returns(Task.FromResult(GetTestSession()));
```

Comment:
Use `ReturnsAsync` instead. It's a bit cleaner. Consider that change all throughout the sample. In this example, the code would become `.ReturnsAsync(GetTestSession());`.